tests/libvmi/storage.go

```diff
@@ -20,7 +20,27 @@
 package libvmi
 
 import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+
 	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/client-go/log"
+	. "kubevirt.io/kubevirt/tests/framework/matcher"
+	"kubevirt.io/kubevirt/tests/libstorage"
```

Comment:
This will not allow `libstorage` package to import `libvmi`, which is more appropriate IMO than the other way around.

Reply:
On this I disagree, it makes more sense the libvmi will need stuff from libstorage and not the other way around, this file specifically is libvmi/storage so it makes sense that it will use storage stuff that are more general and not specific to vmi.

Reply:
so what package to do vmi related operations? I dont think libstorage is the one.. why limit libvmi to only spec? why shouldnt it be anything related to vm?

Reply:
I am not against abstracting operations on VMI/s form this package, but so far I have not seen the need. More particularly, the library is suppose to contain common procedures all (or most) tests need. Specialized (SIG, test groups) tests should create their own libraries.

We have already rejected additions to this library which have been too specific.

For instance, `VerifyVolumeAndDiskVMAdded` is used by a specific group of tests (storage/hotplug), so keeping it there would have been better than sharing it with all tests. All other consumers of libvmi should not be re-compiled and affected by changes to the hotplug tests.
Seems that `VerifyVolumeAndDiskVMIAdded` is used solley by hotplug as well.

`AddVolumeAndVerify` is used by storage/restore and storage/snapshot , which can be split into two parts:
- Building the spec.
- Creating the object.

If you can split these two actions, then moving the build of the spec to libvmi and keeping the creation of the object in the tests (or in libstorage) seems good to me. But I think it requires moving some parts of libstorage to libvmi.

My feeling is that you are trying to move things into libvmi from the "middle". I would start moving all spec related building first, only then proceed with higher level handlers.
My priority of placing handlers is as following:
- In the specific test.
- In the test file.
- In the test topic library (e.g. libnet, libnode, libstorage).
- In common cross tests libraries, e.g. libvmi, console, login.

When the usage crosses boundaries (e.g. tests, files, topics) then it can be promoted up the chain to a more common library.